xcolor: unstable -> 0.5.0

bjpbakker opened the pull request:
Also create a desktop item so xcolor shows up in application launchers.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

bjpbakker committed:
xcolor: unstable -> 0.5.0

Also create a desktop item so xcolor shows up in application launchers.

SuperSandro2000 commented on pkgs/tools/graphics/xcolor/default.nix:
```diff
@@ -1,30 +1,47 @@
+    mkdir -p $out/share/applications
+    cp "${desktopItem}"/share/applications/* $out/share/applications/
+
+    install -D -m644 -- man/xcolor.1 $out/share/man/man1/xcolor.1
```
This should use installManPage

SuperSandro2000 commented on pkgs/tools/graphics/xcolor/default.nix:
```diff
@@ -1,30 +1,47 @@
 
+  postInstall = ''
+    mkdir -p $out/share/applications
+    cp "${desktopItem}"/share/applications/* $out/share/applications/
```
Please use copyDesktopItems instead.

SuperSandro2000 commented on pkgs/tools/graphics/xcolor/default.nix:
```diff
@@ -1,30 +1,47 @@
+    cp "${desktopItem}"/share/applications/* $out/share/applications/
+
+    install -D -m644 -- man/xcolor.1 $out/share/man/man1/xcolor.1
+    install -D -m644 -- extra/icons/xcolor-16.png $out/share/icons/hicolor/16x16/apps/xcolor.png
+    install -D -m644 -- extra/icons/xcolor-24.png $out/share/icons/hicolor/24x24/apps/xcolor.png
+    install -D -m644 -- extra/icons/xcolor-32.png $out/share/icons/hicolor/32x32/apps/xcolor.png
+    install -D -m644 -- extra/icons/xcolor-48.png $out/share/icons/hicolor/48x48/apps/xcolor.png
+    install -D -m644 -- extra/icons/xcolor-256.png $out/share/icons/hicolor/256x256/apps/xcolor.png
+    install -D -m644 -- extra/icons/xcolor-512.png $out/share/icons/hicolor/512x512/apps/xcolor.png
```
The -- are not required and this can be simplified by a lot with a for loop.

SuperSandro2000 commented on pkgs/tools/graphics/xcolor/default.nix:
```diff
@@ -1,30 +1,47 @@
-{ lib, rustPlatform, fetchFromGitHub, fetchpatch, pkg-config, libX11, libXcursor, libxcb, python3 }:
+{ lib, rustPlatform, fetchFromGitHub, fetchpatch, pkg-config, libX11, libXcursor
```
```suggestion
{ lib, rustPlatform, fetchFromGitHub, pkg-config, libX11, libXcursor
```

SuperSandro2000 commented on pkgs/tools/graphics/xcolor/default.nix:
```diff
@@ -1,30 +1,47 @@
+    name = "XColor";
+    exec = "xcolor -s";
+    desktopName = "XColor";
+    comment =
+      "Select colors visible anywhere on the screen to get their RGB representation";
```
```suggestion
    comment = "Select colors visible anywhere on the screen to get their RGB representation";
```

bjpbakker committed:
xcolor: cleanup install phase

Use available utility functions for installing man pages and desktop items. Also
cleanup redundant icon copy lines.

bjpbakker commented:
Thanks for the feedback, wasn't aware of the utility functions yet. Good to have these :)
